lib/xcodeproj/project/object/native_target.rb

```diff
@@ -622,6 +627,22 @@ def sort(_options = nil)
             end
           end
         end
+
+        def to_hash_as(method = :to_hash)
+          hash_as = super
+          if !hash_as['packageProductDependencies'].nil? && hash_as['packageProductDependencies'].empty?
+            hash_as.delete('packageProductDependencies')
+          end
+          hash_as
+        end
+
+        def to_ascii_plist
+          plist = super
+          if !plist.value['packageProductDependencies'].nil? && plist.value['packageProductDependencies'].empty?
+            plist.value.delete('packageProductDependencies')
+          end
+          plist
```

Comment:
is this covered by any tests?

Reply:
it is already covered by existing tests but I might add an explicit one.

Reply:
added explicit tests.